python/plugins/processing/core/parameters.py

```diff
@@ -1186,15 +1190,17 @@ def getAsScriptCode(self):
         param_type = ''
         if self.optional:
             param_type += 'optional '
-        param_type += 'string'
-        return '##' + self.name + '=' + param_type + self.default
+        param_type += 'string '
+        return '##' + self.name + '=' + param_type + str(self.default)
 
     @classmethod
     def fromScriptCode(self, line):
         isOptional, name, definition = _splitParameterOptions(line)
         descName = _createDescriptiveName(name)
         if definition.lower().strip().startswith('string'):
-            default = definition.strip()[len('string') + 1:]
+            default = definition.strip()[len('string') + 1:] or None
+            if default == 'None':
+                default = None
```

Comment:
Hmm... At the moment, the default for the default value is `None` (https://github.com/m-kuhn/QGIS/blob/e41d667faad66c356e4236fc750efe2dc3c9a5c1/python/plugins/processing/core/parameters.py#L1165), so I assumed that it is supported. All that code was left untouched.

To fully (de)serialise `None` values, we will need to somehow add more information to the exported string.

like `None` vs `'None'` which will also require escaping because we start using `'` as control sequence (`'hey it\'s me'`).